File: 2221-find-triangular-sum-of-an-array/2221-find-triangular-sum-of-an-array.cpp
class Solution {
public:
    int triangularSum(vector<int>& nums) {
        int n = nums.size();
        vector<int> v[2];
        v[0] = v[1] = nums;
        for(int i = 1; i < n; i++) {
            for(int j = 0; j < n-i; j++) {
                v[i&1][j] = (v[~i&1][j] + v[~i&1][j+1]) % 10;
            }
        }
        return v[~n&1][0];
    }
};

File: 2221-find-triangular-sum-of-an-array/2221-find-triangular-sum-of-an-array.py
@cache
def C(n, k):
    if k == 0: return 1
    if n <= 0: return 0
    return C(n-1, k-1) + C(n-1, k) % 10

class Solution:
    def triangularSum(self, nums: List[int]) -> int:
        n = len(nums)
        ans = 0
        for i in range(n):
            ans += nums[i] * C(n-1, i) % 10
            ans %= 10
        return ans